src/graphnet/datasets/snowstorm_dataset.py

```diff
@@ -0,0 +1,159 @@
+"""Snowstorm dataset module hosted on the IceCube Collaboration servers."""
+
+import pandas as pd
+import re
+import os
+from typing import Dict, Any, Optional, List, Tuple, Union
+from glob import glob
+from sklearn.model_selection import train_test_split
+
+from graphnet.data.constants import FEATURES, TRUTH
+from graphnet.data.curated_datamodule import IceCubeHostedDataset
+from graphnet.data.utilities import query_database
+from graphnet.models.graphs import GraphDefinition
+
+
+class SnowStormDataset(IceCubeHostedDataset):
+    """IceCube SnowStorm simulation dataset.
+
+    More information can be found at
+    https://wiki.icecube.wisc.edu/index.php/SnowStorm_MC#File_Locations
+    This is a IceCube Collaboration simulation dataset.
+    Requires a username and password.
+    """
+
+    _experiment = "IceCube SnowStorm dataset"
+    _creator = "Severin Magel"
+    _citation = "arXiv:1909.01530"
+    _available_backends = ["sqlite"]
+
+    _pulsemaps = ["SRTInIcePulses"]
+    _truth_table = "truth"
+    _pulse_truth = None
+    _features = FEATURES.SNOWSTORM
+    _event_truth = TRUTH.SNOWSTORM
+    _data_root_dir = "/data/ana/graphnet/Snowstorm_l2"
+
+    def __init__(
+        self,
+        run_ids: List[int],
+        graph_definition: GraphDefinition,
+        download_dir: str,
+        truth: Optional[List[str]] = None,
+        features: Optional[List[str]] = None,
+        train_dataloader_kwargs: Optional[Dict[str, Any]] = None,
+        validation_dataloader_kwargs: Optional[Dict[str, Any]] = None,
+        test_dataloader_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        """Initialize SnowStorm dataset."""
+        self._run_ids = run_ids
+        self._zipped_files = [
+            os.path.join(self._data_root_dir, f"{s}.tar.gz") for s in run_ids
```

Comment:
I think there's currently 28 valid run id's, which is a subset of the overall available sets. I think the current code assumes the user knows this. Perhaps it would be wise to include an assertion that checks that the user-given ids actually exists.